dlib/control/approximate_linear_models_abstract.h

```diff
@@ -128,86 +119,5 @@ namespace dlib
 
 // ----------------------------------------------------------------------------------------
 
-    template <
-        typename feature_extractor
-        >
-    class policy
```

Comment:
I don't really like that approximate_linear_models_abstract.h is so empty now.  The general idea with this file is that someone can read it and understand how to use the module.  It defines the key interfaces they need to implement (i.e. the model/feature extractor class the user needs to make).  It also shows them where it goes in the policy object.  Breaking it up into too many files just makes it hard for people to follow the thread.  

So put the policy objects in this file, the same with the documentation about what exactly constitutes the model interface.